Fixed several issues with Qt5 generator and Qt5 PetStore tests

philicious opened the pull request:
@wing328 as announced in my email, here is the fix for the missing data-type handlers in the template thats rendered into SWGHelpers.cpp

QDate and QDateTime handling was missing entirely. I also found QString handling to be missing in arrays.

PS: There are four overloaded `QString stringValue(..)` functions which are unused and never called as it seems. By logic, they would be missing overloaded funcs for float, double, QDate and QDateTime. I didnt add those tho as they are never called and SDK is working still fine.

wing328 commented:
@philicious thanks for the PR. Can you please also update the Qt5 C++ Petstore client by running `./bin/qt5-petstore.sh` and then run the tests locally (https://github.com/swagger-api/swagger-codegen/tree/master/samples/client/petstore/qt5cpp/PetStore) ?

philicious commented:
@wing328 I did `mvn clean package` and `./bin/qt5-petstore.sh` on my fix-branch, however the generated Qt5-PetStore .h/.cpp didnt change, didnt expect it either as I didnt change generator code. 
(On a side-note, the https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore.yaml doesnt cover all types I think, e.g. its missing `format: date` only has date-time)

So I then went on to `qmake PetStore.pro && make` the sources, however there are a couple of things broken with the Qt5 tests preventing it from building:
- https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/qt5cpp/client/SWGOrder.h#L13 should read `<QDateTime>` , I fear I'm to less into the generator code to understand and fix how it decides when to render local or global includes.
- https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/qt5cpp/client/SWGStoreApi.h#L7 should read `<QMap`>, same as with QDateTime
- https://github.com/swagger-api/swagger-codegen/blob/master/samples/client/petstore/qt5cpp/client/SWGPetApi.cpp#L441 should read `(*file).local_filename` or simply `file->local_filename`
- more errors I didnt go on fixing yet

philicious commented:
wait, I think I found where the includes are generated, https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Qt5CPPGenerator.java#L174-L180 and therefore QMap and QDateTime and QDate would need to be added here https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Qt5CPPGenerator.java#L137

oh well..ye I think I can fix the tests and also the generator. should this all go into this PR or how shall we handle it?

philicious commented:
@wing328 I changed PR title, added some more commits to fix up other issues in generator aswell as in the tests. test is building again and checks do pass

(however I did not extend the test cases. I simply fixed everything to get them build again)

```
********* Start testing of PetApiTests *********
Config: Using QtTest library 5.2.1, Qt 5.2.1
PASS   : PetApiTests::initTestCase()
PASS   : PetApiTests::findPetsByStatusTest()
QDEBUG : PetApiTests::createAndGetPetTest() got a request body
PASS   : PetApiTests::createAndGetPetTest()
QDEBUG : PetApiTests::updatePetTest() got a request body
QDEBUG : PetApiTests::updatePetTest() got a request body
PASS   : PetApiTests::updatePetTest()
QDEBUG : PetApiTests::updatePetWithFormTest() got a request body
PASS   : PetApiTests::updatePetWithFormTest()
PASS   : PetApiTests::cleanupTestCase()
Totals: 6 passed, 0 failed, 0 skipped
********* Finished testing of PetApiTests *********
```

wing328 commented:
The change and test result look good to me. Thanks for the contribution.
